Source/CarthageKit/GitHub.swift

```diff
@@ -265,6 +268,28 @@ private func fetchAllPages(URL: NSURL, credentials: GitHubCredentials?) -> ColdS
 			let thisData = ColdSignal.single(data)
 
 			if let HTTPResponse = response as? NSHTTPURLResponse {
+				if HTTPResponse.statusCode != 200 && HTTPResponse.statusCode != 404 {
+					return thisData
+						.tryMap { data, error -> AnyObject? in
+							return NSJSONSerialization.JSONObjectWithData(data, options: nil, error: error)
+						}
+						.map { dictionary -> String in
+							let message = JSONValue.parse(dictionary)["message"]
+							switch message {
+							case let .Some(.JSONString(message)):
+								return message
+							default:
+								return NSString(data: data, encoding: NSUTF8StringEncoding) ?? "Unknown error requesting \(URL)"
```

Comment:
We could use `+[NSHTTPURLResponse localizedStringForStatusCode:]` for the unknown case.